### PR TITLE
Fix custom writer marks and nodes name

### LINKED ISSUE #5731: [v4] Selecting spesific custom `writerMarks` doesn't work
I'm not sure this is a bug or I'm missing some setup. Selecting spesific custom `writerMarks` doesn't work for writer field.

### Plugin

````js
panel.plugin("my/writer-mark", {
  writerMarks: {
    smile: {
      get button() {
        return {
          icon: "smile",
          label: "Smile"
        }
      }
    }
  }
});
````

### Blueprint
````yaml
fields:
  writer:
    type: writer
    marks:
      - bold
      - smile
    toolbar:
      inline: false
      marks:
        - bold
        - smile
````

### Result
screen-capture-1691-Test - Mægazine-localhost

## Additional Context

I can see the button when `marks: true`

````yaml
fields:
  writer:
    type: writer
    marks: true
    toolbar:
      inline: false
````

screen-capture-1692-Test - Mægazine-localhost

Kirby 4.0.0-beta.2

### PR DESCRIPTION
**Pull request by afbora**
### Fixes
- Fix custom writer marks and nodes name #5731

### Breaking changes
None

## Ready?

- [x] Unit tests for fixed bug/feature
- [x] In-code documentation (wherever needed)
- [x] Tests and checks all pass

### For review team

- [ ] Add changes to release notes draft in Notion
- [x] Add to [website docs release checklist](https://github.com/getkirby/getkirby.com/pulls) (if needed)

### CODE REVIEW

File: panel/.eslintrc.js [hunk 1]
```diff
@@ -1,5 +1,8 @@
 module.exports = {
 	extends: ["eslint:recommended", "plugin:vue/recommended", "prettier"],
+	parserOptions: {
+		ecmaVersion: 2021
+	},
```
Comment (afbora) [inline panel/.eslintrc.js]: Not sure about this change. I have to use version 2021 (currently 2020) to use null coalescing assignment operator `??=` in js.

Comment (bastianallgeier) [reply panel/.eslintrc.js]: If this does not break anything else, I'm happy about it. We could definitely use this in more places.

Comment (distantnative) [reply panel/.eslintrc.js]: Is this automatically compiling backwards on build - or are we raising browser requirements with this?

File: panel/src/components/Forms/Writer/Writer.vue [hunk 2]
```diff
@@ -252,12 +252,15 @@ export default {
...
+			// take each extension object and turn
 			// it into an instance that extends the Mark class
 			for (const name in plugins) {
+				const plugin = plugins[name];
+				plugin.name ??= name;
+
 				marks[name] = Object.create(
 					Mark.prototype,
-					Object.getOwnPropertyDescriptors(plugins[name])
+					Object.getOwnPropertyDescriptors(plugin)
```
Comment (distantnative) [inline panel/src/components/Forms/Writer/Writer.vue]: This is actually exactly what I had in mind, just didn't realize this is already the right spot when I wrote from my phone.

Might have written is just as 

```suggestion
					Object.getOwnPropertyDescriptors({ name, ...plugins[name] })
```

to make it a one-liner change

Comment (afbora) [reply panel/src/components/Forms/Writer/Writer.vue]: That's what I need 💯

**Review by distantnative — APPROVED** [review]

**Pull request merged**